src/main/java/com/savitoh/casacodigoapi/service/AutorService.java

```diff
@@ -0,0 +1,44 @@
+package com.savitoh.casacodigoapi.service;
+
+import com.savitoh.casacodigoapi.model.Autor;
+import com.savitoh.casacodigoapi.repository.AutorRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+
+@Service
+@Transactional(readOnly = false)
```

Comment:
*Usamos tudo que conhecemos que está pronto. Só fazemos código do zero se for estritamente necessário.*

Mesmo que essa classe fosse ficar... O readOnly é false por default. Tudo que você escreve vai querer ser interpretado por alguém.. E quanto mais coisa precisa ser interpretada, mais complicado é o código.